### DISCUSSION TITLE
Release API and conditional requests

### DISCUSSION THREAD

**Discussion by amirmalka** [community/community#33054] [Apps, API and Webhooks]
I am trying to find out whether a new release has been published for a specific repository with a conditional request.

The API I use is: `https://api.github.com/repos/<OWNER>/<REPO>/releases/latest` (https://docs.github.com/en/rest/releases/releases#get-the-latest-release) 

I'm setting the `If-Modified-Since` & `If-None-Match` headers on the request (with their respective values), however, I get a 200 response for every response even when no changes were made, any idea what am I doing wrong?

Thanks!

**Comment by jxmot** [comment]
Have you looked at the content of the response? I just tried the same endpoint without the conditionals on one of my **public** repositories and it worked as expected. I get the data for the most recent release which includes the date it was created.

Please note that the documentation regarding the response **might be wrong** or not show the complete conditions under which it was obtained. Unfortunately that important information is missing. For example, in the example response the `"assets"` array contains data. Although it's not documented anywhere I believe that the "release" used in the example had additional resources uploaded to the release. 

Try removing the conditionals and check the response data. It should contain all of the info for the latest release. However, if the repository you're targeting is _private_ then you will need to supply a token in the header under "Authorization". For public repositories the token is not required.

Also, the conditionals your using may not be supported by the endpoint. The GitHub docs for that endpoint do not make any mention of them. 

What is the result you're looking for when using the conditionals?

**Reply by amirmalka** [reply]
I was trying to check for a latest release, so that my app will not hit GitHub's request limit (https://docs.github.com/en/rest/overview/resources-in-the-rest-api#conditional-requests)

Anyways, it looks like the response supported by this endpoint is only 200

**Reply by jxmot** [reply]
I just tried both on one of my public repositories without any authorization token and it works as described in the link you provided. Both returned the 304. I'm using "Postman" to send requests to the endpoint. A tool like that is very helpful when developing and testing code that sends requests to an API.

If you are concerned about hitting the request limit then you might want to consider adding authorization to your requests. The limit is considerably higher than without it.

You may also want to examine how often your application is making requests. Some may not require frequent calls because the data that is returned doesn't change often. And the responses from those could be cached by your application... depending on if it's a client-side or server-side application.

**Reply by Linka21** [reply]
Maaf saya masih baru menggunakan github bagaimana saya bisa mendapatkan https://github.com/samsesh/SocialBox-Termux.git
